### ISSUE TITLE
Using the new callable notation should be disallowed, even between quotes, if python < 3.11

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#3016]
**Describe the bug**
It turns out CPython actually evaluates the expression within quotes when it is passed to a `bound` parameter of a `TypeVar`, which causes a crash in runtime when using typing syntax that is not yet supported.

**To Reproduce**
Try to execute the following with python < 3.11 (tested with 3.9.7)

```python3
#pyright: strict

from typing import TypeVar

T = TypeVar("T", bound="() -> int") # no static check errors
```

this crashes with `SyntaxError: Forward reference must be an expression -- got '() -> int'`

**Expected behavior**
I guess the correct behavior is to disallow new typing features within quotes, at least for the `bounds` parameter, but potentially in other places too.

**VS Code extension or command-line**
extension version 1.1.220

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.221, which I just published. It will also be included in the next release of pylance.
